### ISSUE TITLE
OBS Studio>24.0.3 use Game Capture cause Minecraft 1.8.9 crash

### ISSUE THREAD

**Issue by guation** [obsproject/obs-studio#5841]
### Operating System Info

Windows 10

### OBS Studio Version

27.2.0 Beta 3

### OBS Studio Log URL

https://obsproject.com/logs/4leKaRa3y4pLTEzq

### Expected Behavior

I want to switch between window and full screen in minecraft.

### Current Behavior

I switch the window to full screen and my minecraft will crash.

### Steps to Reproduce

1.Run Minecraft 1.8.9
2.Open OBS -> sources -> add -> Game Capture -> Mode Capture specific windows -> Windows [java.exe]: Minecraft 1.8.9
3.Back to minecraft, press F11 to make minecraft full screen.
4.Minecraft will crash

### Anything else we should know?

I use a notebook. The built-in display is connected to gpu0, and the external display is connected to gpu1.
Built in display + game capture full screen minecraft crash. （OBS>24.0.3）
Built in display + game capture full screen minecraft does not crash. （OBS=24.0.3）
External display + game capture full screen minecraft does not crash
Built in display + window capture full screen minecraft does not crash.
OS: Windows 10 21H2
CPU: i5-8400
GPU: Intel(R) UHD Graphics 630 (GPU0) + NVIDIA GeForce GTX 1050 Ti (GPU1)

**Comment by guation** [comment]
> 这感觉就像在官方 OBS Discord 服务器中可以轻松解决的问题，真的。
This problem has existed since OBS version 25.0.0

**Comment by Xorshasia** [comment]
Doesn't change the fact it could easily be solved on the Discord server.

**Comment by dodgepong** [comment]
This sounds more likely to be a support issue than a bug. Please submit this through one of our support channels ([Discord server](https://obsproject.com/discord) or the [forums](https://obsproject.com/forum), see more at our [help page](https://obsproject.com/help)), and if it's determined that this is indeed a bug, then we will re-open this issue.

**Comment by RytoEX** [comment]
I can confirm the described behavior with OBS Studio 27.1.3 and 27.2.0-beta3 with respect to Minecraft 1.8.9.  I can confirm that OBS Studio 24.0.3 works while 25.0.0-rc2 causes Minecraft to crash.  I have not yet done a bisect, and I have not checked other versions of Minecraft.

**Comment by guation** [comment]
> 我可以使用 OBS Studio 27.1.3 和 27.2.0-beta3 确认所描述的关于 Minecraft 1.8.9 的行为。我可以确认 OBS Studio 24.0.3 可以工作，而 25.0.0-rc2 会导致 Minecraft 崩溃。我还没有完成对分，也没有检查过其他版本的 Minecraft。

I tested all minecraft releases in OBS 27.1.3. From 1.0 to 1.12.2 will crash. These versions are really full screen and the mouse can't move out of the game window. From 1.13 to 1.18.1 will not crash. These versions are pseudo full screen, and the mouse can move out of the game window.

**Comment by flaeri** [comment]
Cannot reproduce on single GPU desktop machine. Enabling/disabling fullscreen 3-4 times, and no crash.
![image](https://user-images.githubusercontent.com/50419942/150672894-bb881912-e857-4662-97f1-8b6a9d362ccd.png)
<https://obsproject.com/logs/995BtrImcGWSh2R7>

I am of the opinion that this is primarily an issue on dual GPU system (both active).
Looking at the logs, it seems to me like you have one of your monitors connected to the iGPU of your computer. I believe that if you move all monitors to the discrete GPU, and reboot your computer, the problem will go away.

**Comment by guation** [comment]
> 

I also understand that this is a problem with laptops using hybrid graphics, but my laptop manufacturer does not allow me to connect the built-in display to a separate graphics card. When I connected the monitor externally, my monitor was connected to a separate graphics card. All this was normal, but now I have to use my built-in monitor. This problem does not exist in OBS 24.0.3. This problem began with OBS 25.0.0-rc1, so I think it is a bug.

**Comment by flaeri** [comment]
Yes, you're quite right. I just figured that I would mention this for others that might not be aware.
The log you posted was not from a laptop/all-in-one, it was from a proper desktop platform, so I thought it was worth mentioning.

**Comment by guation** [comment]
> Yes, you're quite right. I just figured that I would mention this for others that might not be aware. The log you posted was not from a laptop/all-in-one, it was from a proper desktop platform, so I thought it was worth mentioning.
> 
> I'm not trying to say this is not a problem, just clarifying the scope, and potentially helping others that come across this issue.

My computer is a Hasee zx6cp5s1 (blue sky nb50) notebook, which uses the CPU of the desktop platform.

**Comment by RytoEX** [comment]
A git bisect points to https://github.com/obsproject/obs-studio/commit/3e3521b9e170ff92df9be15d6757057549f2888d as the bad commit.  It seems like `handle` may become null or invalid when making Minecraft fullscreen, so `capture_alive` may be returning false.  Then I get a read access violation (nullptr) in d3d11-capture.cpp in `d3d11_copy_texture` for some reason.

That's as much as I can figure out.
